diofant/core/tests/test_cache.py

```diff
@@ -96,3 +98,20 @@ def test_nocache(clear_imports, monkeypatch):
     y = Symbol('y')
     result = s.subs(sqrt(x**2), y)
     assert result == 1/y
+
+    # problem from https://groups.google.com/forum/#!topic/sympy/LkTMQKC_BOw
+    # see commit c459d18
+    a = Symbol('a', positive=True)
+    f = exp(x*(-a - 1))
+    g = sinh(x)
+    r = f*g  # not raises
```

Comment:
@Ruediger-Braun, you may be interested in fix for the problem, mentioned in https://groups.google.com/forum/#!topic/sympy/LkTMQKC_BOw

This issue was fixed in Diofant 0.8.  By the way, it seems to be fixed in the recent SymPy as well.